### PR TITLE
PAINTROID-439 Re-Implement the drawing surface and basic drawing in Flutter

### PR DESCRIPTION
**Pull request by ikrsk**
- A simple drawing brush and surface
- Top and bottom app bars with Material 3 style

> First PR! Leaving my mark here :D Excited to see this app grow

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Paintroid
+=========
+
+Paintroid, also known as **Pocket Paint**, is associated to [Catroid](https://github.com/Catrobat/Catroid). It is a graphical paint editor application for the Android platform that, among others, allows setting parts of pictures to transparent.
+
+Since Pocket Paint is now available in **Google Play store** you can also download Paintroid (Pocket Paint) from [here](https://play.google.com/store/apps/details?id=org.catrobat.paintroid). Alternatively, you can find it on **F-Droid** [here](https://f-droid.org/packages/org.catrobat.paintroid/).
+
+
+For more information oriented towards developers please visit our [developers page](http://developer.catrobat.org/).
```
Comment (juliajulie95) [inline README.md]: Please mention here that this repo is actually a flutter rewrite of the other repo and link the other repo :)

File: android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        targetSdkVersion flutter.targetSdkVersion
+        versionCode flutterVersionCode.toInteger()
+        versionName flutterVersionName
+    }
+
+    buildTypes {
+        release {
+            // TODO: Add your own signing config for the release build.
+            // Signing with the debug keys for now, so `flutter run --release` works.
+            signingConfig signingConfigs.debug
```
Comment (juliajulie95) [inline android/app/build.gradle]: please delete all the comments here if they are not needed

File: android/app/src/main/res/drawable-v21/launch_background.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+<!-- Modify this file to customize your launch splash screen -->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="?android:colorBackground" />
+
+    <!-- You can insert your own image assets here -->
+    <!-- <item>
+        <bitmap
+            android:gravity="center"
+            android:src="@mipmap/launch_image" />
+    </item> -->
```
Comment (juliajulie95) [inline android/app/src/main/res/drawable-v21/launch_background.xml]: delete if not used

File: android/app/src/main/res/drawable/launch_background.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<!-- Modify this file to customize your launch splash screen -->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@android:color/white" />
+
+    <!-- You can insert your own image assets here -->
+    <!-- <item>
+        <bitmap
+            android:gravity="center"
+            android:src="@mipmap/launch_image" />
+    </item> -->
```
Comment (juliajulie95) [inline android/app/src/main/res/drawable/launch_background.xml]: delete

File: lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        useMaterial3: true,
+        colorScheme: ColorScheme.fromSeed(
+          seedColor: const Color(0xFF0097A7),
+          brightness: Brightness.light
+        ),
+      ),
+      darkTheme: ThemeData.from(
+        useMaterial3: true,
+        colorScheme: ColorScheme.fromSeed(
+          seedColor: const Color(0xFF00838F),
```
Comment (juliajulie95) [inline lib/main.dart]: Could we add the colors to a file to have them all stored together and only reference them here?

File: lib/tool/brush_tool.dart [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  @override
+  void onDrag(Offset point) {
+    pathToDraw.lineTo(point.dx, point.dy);
+  }
+
+  @override
+  void onUp(Offset? point) {
+    // This basically makes the path a point.
```
Comment (juliajulie95) [inline lib/tool/brush_tool.dart]: I think we do not need the comment, this makes sense without it :D

File: lib/ui/pocket_paint.dart [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+import 'drawing_board.dart';
+
+class PocketPaint extends StatelessWidget {
+  final String title;
+
+  const PocketPaint({Key? key, required this.title}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (juliajulie95) [inline lib/ui/pocket_paint.dart:11]: please already add the colors and correct icons for the appbar and bottom bar, make it look the same as in the current app

File: lib/ui/drawing_board.dart [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    return GestureDetector(
+      onPanDown: (details) {
+        setState(() => brushTool.onDown(details.localPosition));
+      },
+      onPanUpdate: (details) {
+        setState(() => brushTool.onDrag(details.localPosition));
+      },
+      onPanEnd: (_) {
+        setState(() => brushTool.onUp(null));
```
Comment (juliajulie95) [inline lib/ui/drawing_board.dart:43]: Could we used some kind of provider instead of the setState()?

**Review by juliajulie95 — COMMENTED** [review]
Hi! I added some suggestions, please have a look

File: lib/main.dart (hunk 5 above)
Comment (ikrsk) [reply lib/main.dart]: Yes, that would be better. On it.

**Review by juliajulie95 — APPROVED** [review]
Approved, thank you! :)

**Review by juliajulie95 — APPROVED** [review]

**Pull request merged**